File: src/main/java/com/example/mediaApp/model/entity/GroupRequest.java
package com.example.mediaApp.model.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity(name = "GroupRequest")
@Table(name = "group_request")
@Getter
@Setter
@NoArgsConstructor
@ToString(onlyExplicitlyIncluded = true, includeFieldNames = false)
public class GroupRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    // Bidirectional
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "group_id")
    private Group group;

    // Unidirectional
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "sender_user_id")
    private AppUser senderUser;
}

File: src/main/angular-media-app/src/app/models/group-request.ts
import { Group } from "./group";
import { User } from "./user";

export class GroupRequest {
    id: number;
    group: Group | null;
    senderUser: User | null;

    constructor(){
        this.id = 0;
        this.group = null;
        this.senderUser = null;
    }
}
